packages/next/src/server/web/spec-extension/adapters/headers.ts

```diff
@@ -26,11 +28,13 @@ export class HeadersAdapter extends Headers {
     super()
 
     this.headers = new Proxy(headers, {
-      get(target, prop, receiver) {
+      get(target, prop) {
         // Because this is just an object, we expect that all "get" operations
         // are for properties. If it's a "get" for a symbol, we'll just return
         // the symbol.
-        if (typeof prop === 'symbol') return Reflect.get(target, prop, receiver)
+        if (typeof prop === 'symbol') {
+          return ReflectAdapter.get(target, prop, headers)
```

Comment:
Why remove the `receiver` param? Here, `headers === target` are always the same.

Additionally, without the correct `receiver`, sets on prototype children will install properties on parents, and getters/setters will invoke on the wrong object.